app/controllers/locations_controller.rb

```diff
@@ -8,7 +8,8 @@ class LocationsController < ApplicationController
   layout 'full_width_with_breadcrumbs', only: %i(show index)
 
   def index
-    return render :search unless @postcode.present?
+    return render :search if @postcode.nil?
+    return render :empty_postcode if @postcode.empty?
```

Comment:
Maybe we could put a client validation pattern or something on the frontend to stop this roundtrip for a large number of cases too?